### PR TITLE
Added new environment variable to tests - TESTS_SKIP_BEFORE

### PR DESCRIPTION
**Pull request by pranavrth**

### CODE REVIEW

**Comment by milindl** [comment]
It causes an unclean exit for me while using ASaN. The error code makes it seem like SIGPIPE
```
$ ./dev-conf.sh asan
$ cd tests && /* set up trivup/brokers */
$ TESTS_SKIP_BEFORE=0137 ./run-test.sh
```

**Comment by pranavrth** [comment]
> It causes an unclean exit for me while using ASaN. The error code makes it seem like SIGPIPE
> 
> ```
> $ ./dev-conf.sh asan
> $ cd tests && /* set up trivup/brokers */
> $ TESTS_SKIP_BEFORE=0137 ./run-test.sh
> ```

For me this is passing

```
| 0134_ssl_provider                        |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0135_sasl_credentials                    |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0136_resolve_cb                          |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0137_barrier_batch_consume               |     PASSED |  59.579s |
| 0138_admin_mock                          |     PASSED |   0.034s |
| 0139_offset_validation_mock              |     PASSED |   5.642s |
| 0140_commit_metadata                     |     PASSED |   0.513s |
#==================================================================#
[<MAIN>                      / 60.192s] 0 thread(s) in use by librdkafka
[<MAIN>                      / 60.192s]
============== ALL TESTS PASSED ==============
###
###  ./test-runner in bare mode PASSED! ###
###
[TRIVUP:KafkaCluster@3.3.0] bash-5.2$
```

**Comment by pranavrth** [comment]
Maybe something in Linux. I will try to check there as well.

**Comment by pranavrth** [comment]
Same in Linux.

```
| 0122_buffer_cleaning_after_rebalance     |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0123_connections_max_idle                |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0124_openssl_invalid_engine              |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0125_immediate_flush                     |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0126_oauthbearer_oidc                    |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0128_sasl_callback_queue                 |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0129_fetch_aborted_msgs                  |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0130_store_offsets                       |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0131_connect_timeout                     |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0132_strategy_ordering                   |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0133_ssl_keys                            |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0134_ssl_provider                        |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0135_sasl_credentials                    |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0136_resolve_cb                          |    SKIPPED |   0.000s | ignoring test before TESTS_SKIP_BEFORE
| 0137_barrier_batch_consume               |     PASSED |  69.719s |
| 0138_admin_mock                          |     PASSED |   0.029s |
| 0139_offset_validation_mock              |     PASSED |   5.705s |
| 0140_commit_metadata                     |     PASSED |   6.245s |
#==================================================================#
[<MAIN>                      / 70.020s] 0 thread(s) in use by librdkafka
[<MAIN>                      / 70.020s] 
============== ALL TESTS PASSED ==============
###
###  ./test-runner in bare mode PASSED! ###
###
[TRIVUP:KafkaCluster@3.3.0] 
```

**Comment by milindl** [comment]
Sure, let's syncup to merge this, I took it a bit further; the fail only occurs when sqlite3 isn't installed (linux). But TESTS=140 works, TESTS_SKIP_BEFORE=0140 doesn't.

**Comment by pranavrth** [comment]
I don't even have sqllite3 package in my linux machine.

File: tests/README.md [hunk 1]
```diff
@@ -186,6 +186,10 @@ be it `make`, `run-test.sh`, `until-fail.sh`, etc.
                           with `TESTS=0000`.
                           See [../src/rdunittest.c](../src/rdunittest.c) for
                           unit test names.
+ * `TESTS_SKIP_BEFORE=0nnn` - skip tests before this test. Tests are skipped
+                              even if they are part of `TESTS` variable.
+                              Usage: `TESTS_SKIP_BEFORE=0030`. All the tests
+                              till test 0030 is skipped.
```
Comment (milindl) [inline tests/README.md:192]:
```suggestion
                              until test 0030 are skipped.
```

**Review by milindl — APPROVED** [review]
lgtm, thanks for this, one minor comment,

**Pull request merged**